### PR TITLE
fix(ButtonWithOptions): (#1209) : Limiting option to one required child

### PR DESCRIPTION
**Pull request by Wo1v3r**
## Before you request a review, please make sure to check the following boxes:

- [X] Explain what has changed.
- [X] Explain why the change is needed.
- [X] Added tests.
- [X] Update documentation.
- [ ] Merge master with no conflicts.
- [x] Make sure the build is passing on ci.
- [x] If this pr closes any issue, please add a github link to close it.
- [x] If you changed something in the product design/UX, you got the approval of [Ben Benhorin](https://wix.slack.com/messages/@benb).

## ButtonWithOptions fix:
TLDR: Derives from issue #1209: Error originated from DropDownLayout validating option children elements, Lifted the error check through propTypes to ButtonWithOptions.Option

- Corrected ButtonWithOptions Story, passing only one child to Option
- Constrained 1 required child for ButtonWithOption.Option with PropTypes (Validation tested)
- Added that constraint to ButtonWithOption documentation
- Corrected the error thrown from DropDownLayout to mark that it originated from this component.

### CODE REVIEW

File: src/ButtonWithOptions/ButtonWithOptions.spec.js [hunk 1]
```diff
@@ -27,12 +27,33 @@ const runButtonWithOptionsTest = driverFactory => {
...
-        </ButtonWithOptions.Button>
-
+      Test
+      </ButtonWithOptions.Button>
         {optionsArray}
-      </ButtonWithOptions>
-    );
+      </ButtonWithOptions>);
+
+    describe('Option children validation', () => {
```
Comment (Wo1v3r) [inline src/ButtonWithOptions/ButtonWithOptions.spec.js]: Move the describe to end of unit test suite, irrelevant from usual test flow

File: src/ButtonWithOptions/ButtonWithOptions.spec.js [hunk 2]
```diff
@@ -27,12 +27,33 @@ const runButtonWithOptionsTest = driverFactory => {
     const buttonWithOptions = props => (
       <ButtonWithOptions {...props}>
         <ButtonWithOptions.Button {...props}>
-          Test
-        </ButtonWithOptions.Button>
-
+      Test
```
Comment (Wo1v3r) [inline src/ButtonWithOptions/ButtonWithOptions.spec.js]: Move that back to its place :3 .

File: src/ButtonWithOptions/ButtonWithOptions.js [hunk 3]
```diff
@@ -166,5 +164,26 @@ ButtonWithOptions.Button = props =>
...
 
+ButtonWithOptions.Option =
+class Option extends React.Component {
+
+  static displayName = 'ButtonWithOptions.Option';
+
+  static propTypes = {
+    children: (props, propName, componentName) => {
+      const prop = props[propName];
+      if (!prop) {
```
Comment (lbelinsk) [inline src/ButtonWithOptions/ButtonWithOptions.js]: no need for this one, since the second condition here covers it. `Should have a single child` is good enough even no child is passed

File: src/ButtonWithOptions/ButtonWithOptions.spec.js (hunk 2 above)
Comment (lbelinsk) [reply src/ButtonWithOptions/ButtonWithOptions.spec.js]: What? why is this not aligned like it was before? it needs more padding

Comment (Wo1v3r) [reply src/ButtonWithOptions/ButtonWithOptions.spec.js]: prolly my ide ill fix that :]

File: src/ButtonWithOptions/ButtonWithOptions.js (hunk 3 above)
Comment (Wo1v3r) [reply src/ButtonWithOptions/ButtonWithOptions.js]: cool ill omit that
